src/components/taskDetails/Details.tsx

```diff
@@ -5,16 +5,63 @@ import { TaskDetailsProps } from '@/interfaces/taskDetails.type';
 import extractRepoName from '@/utils/extractRepoName';
 import Link from 'next/link';
 import { FaReceipt } from 'react-icons/fa6';
-import DevFeature from '../DevFeature';
+import moment from 'moment';
+import Tooltip from '@/components/common/Tooltip/Tooltip';
+import { useRouter } from 'next/router';
 
 const Details: FC<TaskDetailsProps> = ({ detailType, value, url }) => {
     const color = value ? setColor?.[value] : undefined;
     const isGitHubLink = detailType === 'Link';
     const gitHubIssueLink = isGitHubLink ? value : undefined;
+    const router = useRouter();
+    const isDevMode = router.query.dev === 'true' ? true : false;
+
+    const getRelativeTime = (timestamp: string | number | undefined) => {
+        return timestamp ? moment(timestamp).fromNow() : 'N/A';
+    };
+
+    const getTooltipText = (timestamp: string | number | undefined) => {
+        return timestamp ? moment(timestamp).format('LLLL') : 'N/A';
+    };
+
+    const isPastDate = (timestamp: string | number | undefined) => {
+        return timestamp ? moment(timestamp).isBefore(moment()) : false;
+    };
+
+    const getHumanReadableTime = (timestamp: string | number | undefined) => {
+        if (!timestamp) return 'N/A';
+
+        const now = moment();
+        const futureTimestamp = moment(timestamp);
+
+        if (futureTimestamp.isAfter(now)) {
+            return `in ${moment
+                .duration(futureTimestamp.diff(now))
+                .humanize()}`;
+        } else {
+            return `${moment
+                .duration(now.diff(futureTimestamp))
+                .humanize()} ago`;
+        }
+    };
+
+    const isTimeDetail =
+        detailType === 'Started On' || detailType === 'Ends On';
+
+    const renderedValue = value ?? 'N/A';
+    const formattedDetailType = isDevMode
+        ? detailType === 'Started On'
```

Comment:
please avoid using. the nested ternary operator.

Reply:
I've decided to keep the current structure for simplicity and consistency with the existing codebase. The nested ternary operator is still clear and concise for this particular context. That's why I retained the nested ternary operator for simplicity.